core/src/test/scala/weaponregex/parser/ParserTest.scala

```diff
@@ -0,0 +1,20 @@
+package weaponregex.parser
+
+import weaponregex.model.regextree.RegexTree
+
+import scala.util.Failure
+
+abstract class ParserTest extends munit.FunSuite {
```

Comment:
This is great to have some shared logic for the tests! I noticed that the two test suites also have a lot of the same tests. Can you move those into the shared abstract class too? That way there is less duplication, like you are already doing with the `parseErrorTest`

Reply:
We sure will! We planned on having the tests reorganized after we finished with all the current stuffs.